src/components/Articles/Articles.jsx

```diff
@@ -1,20 +1,31 @@
 import axios from 'axios';
+import { useParams } from "react-router";
 import { useEffect, useState } from 'react';
 import ArticleCard from './ArticleCard';
+import requests from '../../utils/requests';
+
 import './Articles.css';
 
 const Articles = () => {
+    const {topic} = useParams();
     const [articles, setArticles] = useState();
     const [isLoading, setIsLoading] = useState(true);
 
     useEffect(() => {
         setIsLoading(true);
-        axios.get('https://tomnews.herokuapp.com/api/articles')
-        .then(({data}) => {
-            setArticles(data.articles);
-            setIsLoading(false)
+        requests.getArticles()
+        .then((res) => {
+            if (topic !== 'allArticles'){
+                setArticles(res.filter((article) => {
+                    return article.topic == topic;
+                }))
+            }else{
+                setArticles(res);
```

Comment:
I would like you to filter by using a request rather than filtering using your own filtering of the full list of articles. Axios allows you to do API requests with parameter options and it really helpfully will not count the parameter if it undefined when the request is made, e.g. if topic does not exist. Please use that instead

Reply:
When you use axios parameters this will change the usefulness of having 'allArticles' in the topic parametric endpoint as it might break the query